ID-B-20

RDMAI opened the pull request:
Все по 20 заданию.

Перенес всё связанное с Dapper в слой Infrastructure. Не то чтобы меня напрягала зависимость слоя Application от Dapper, просто кажется неправильным писать sql запросы прямо в слое с бизнес логикой

RDMAI committed:
Added GetSpecies and GetBreeds features.
Moved Dapper related logic to Infrastructure layer. Removed dependency for Dapper in Application layer.
Added pagination and sorting helpers for Dapper.

RDMAI committed:
Changed AddPet feature to check for breed existance with new Dapper DBReader

RDMAI committed:
Added DeleteBreed and DeleteSpecies features.

KirillSachkov commented:
> Все по 20 заданию.
> 
> Перенес всё связанное с Dapper в слой Infrastructure. Не то чтобы меня напрягала зависимость слоя Application от Dapper, просто кажется неправильным писать sql запросы прямо в слое с бизнес логикой

Это нормально, так делают во многих коммерческих проектах. Иначе будет слишком много асбтракций над абстракциями

KirillSachkov commented on PetFamily.Backend/src/PetFamily.Infrastructure/DataBaseAccess/Read/DBReaders/VolunteerAggregateDBReader.cs:
```diff
@@ -0,0 +1,146 @@
+using System.Text;
+
+namespace PetFamily.Infrastructure.DataBaseAccess.Read.DBReaders;
+public class VolunteerAggregateDBReader : IVolunteerAggregateDBReader
```
когда мы делаем запросы на чтение, то мы вообще не привязываемся ни к каким агрегатам и тд. запросы на чтение не содержать логики, мы просто берём данные из бд. поэтому название класса как минимум неверное. я бы каждый запрос делал в своей фичи, потому что каждый запрос хочет получить уникальные свои данные

RDMAI committed:
Got rid of the dbreader abstractions.
Also fixed a bug, where the name of the property in sorting needed to match the case of the property name. Now it is case-insensitive.

RDMAI commented:
Выпилил абстракции.

Наверное меня больше всего в этом напрягает возможность смены провайдера базы данных (на прошлой работе переходили с DB2 на MS SQL Server, намучались ребята тогда знатно). SQL у postgres и ms sql, например, мало, но отличается. Как-то странно менять слой Application при неизменной бизнес логике. С другой стороны, какая разница, где менять, если все равно менять.

У тебя, кстати, были такие случаи в практике? Интересно как с этим разные люди разбираются

KirillSachkov commented:
> Выпилил абстракции.
> 
> Наверное меня больше всего в этом напрягает возможность смены провайдера базы данных (на прошлой работе переходили с DB2 на MS SQL Server, намучались ребята тогда знатно). SQL у postgres и ms sql, например, мало, но отличается. Как-то странно менять слой Application при неизменной бизнес логике. С другой стороны, какая разница, где менять, если все равно менять.
> 
> У тебя, кстати, были такие случаи в практике? Интересно как с этим разные люди разбираются

Таких случаев в практике особо не было, на самом деле orm редко меняют, а если меняют, то это в любом случае будет геморой, сколько бы абстракций ты не делал
